lte/gateway/c/li_agent/ProxyConnector.cpp

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <iostream>
+
+#include <libmnl/libmnl.h>
+#include <linux/netfilter/nfnetlink.h>
+#include <linux/netfilter/nfnetlink_conntrack.h>
+
+#include <linux/if_packet.h>
+#include <string.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+#include <net/if.h>
+#include <netinet/ether.h>
+#include <linux/ip.h>
+#include <memory>
+#include <pcap.h>
+
+#include <openssl/ssl.h>
+#include <openssl/err.h>
+
+#include "ProxyConnector.h"
+
+#include "magma_logging.h"
+
+namespace magma {
+namespace lte {
+
+ProxyConnector::ProxyConnector(
+    const std::string& proxy_addr, const int proxy_port,
+    const std::string& cert_file, const std::string& key_file)
+    : proxy_addr_(proxy_addr),
+      proxy_port_(proxy_port),
+      cert_file_(cert_file),
+      key_file_(key_file) {
+  ssl_ = GetSSLSocket();
+}
+
+void ProxyConnector::LoadCertificates(SSL_CTX* ctx) {
+  if (SSL_CTX_use_certificate_file(ctx, cert_file_.c_str(), SSL_FILETYPE_PEM) <=
+      0) {
+    ERR_print_errors_fp(stderr);
+    abort();
```

Comment:
Preference is to return a Status from any failure like this - as abort() removes the option of the caller to decide how critical this failure is.

Reply:
I've converted to just -1 on errors, it would be nice if we adapt a Result class/struct across all services as you mentioned in the task.